src/internal/config.ts

```diff
@@ -7,5 +7,15 @@ export const config = {
    * The promise constructor used by default for methods such as
    * {@link toPromise} and {@link forEach}
    */
-  Promise
+  Promise,
+
+  /**
+   * If true, turns on synchronous error rethrowing, which is a deprecated behavior
+   * in v6 and higher. This behavior enables bad patterns like wrapping a subscribe
+   * call in a try/catch block. It also enables producer interference, a nasty bug
+   * where a multicast can be broken for all observers by a downstream consumer with
+   * an unhandled error. DO NOT USE THIS FLAG UNLESS IT'S NEEDED TO BY TIME
+   * FOR MIGRATION REASONS.
```

Comment:
honestly I'd rather left this as undocumented, but guess that'll bring another confusion too. At least let's add `//TODO` note to move this into new `compat` pkg in further.

Reply:
I think the compat package might be able to just turn this on. cc @jasonaden

Reply:
then again... this changes behavior, where the compat stuff just has some performance and payload size changes associated with using it. So it's probably better to have people explicitly opt into this behavior.

Reply:
What I meant was moving this config itself into compat pkg later, so anyone want to opt in need compat instead of core directly exposes this. I believe eventually core shouldn't contain this _flag_ at least.